main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala

```diff
@@ -424,7 +424,11 @@ object WeededAst {
 
   case class HandlerRule(op: Name.Ident, fparams: List[FormalParam], exp: Expr)
 
-  case class WithHandler(eff: Name.QName, rules: List[HandlerRule])
+  sealed trait Handler
+
+  case class WithHandler(eff: Name.QName, rules: List[HandlerRule]) extends Handler
+
+  case class RunHandler(exp: Expr) extends Handler
```

Comment:
These should be inside an `object Handler`. I would also consider renaming them:

```
RunHandler -> RunWithHandler
WithHandler -> InlineHandler (?)
```